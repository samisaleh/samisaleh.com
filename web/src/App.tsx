import React from 'react';
import { BrowserRouter, Route, Switch, Redirect } from 'react-router-dom';
import { ReactElement } from 'react';
import { Home } from './pages/home';
import { pageGroups } from './pages';

export default function App(): ReactElement {
    return (
        <BrowserRouter>
            <Switch>
                <Route exact path="/" component={Home} />
                {pageGroups(true).map(({ pages, redirects = [] }) => {
                    return [
                        ...redirects.map(({ from, to }) => <Redirect key={`${from}-${to}`} from={from} to={to} />),
                        ...pages.map(page => <Route key={page.path} path={page.path} component={page.component} />),
                    ];
                })}
                <Route component={Home} />
            </Switch>
        </BrowserRouter>
    );
}
