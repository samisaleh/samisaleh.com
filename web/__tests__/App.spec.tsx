import React from 'react';
import App from '../src/App';
import { create } from 'react-test-renderer';

describe('<App />', () => {
    it('Matches snapshot', () => {
        const app = create(<App />);
        expect(app.toJSON()).toMatchSnapshot();
    });
});
