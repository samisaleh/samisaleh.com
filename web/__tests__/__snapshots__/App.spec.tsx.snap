// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Matches snapshot 1`] = `
<div>
  <div
    className="homeBanner"
  >
    <div
      className="ui container"
    >
      <h1
        className="ui inverted header"
      >
        SamiSaleh.com
      </h1>
    </div>
  </div>
  <div
    className="ui container"
  >
    <div
      className="ui basic segment"
    >
      <div
        className="ui centered stackable center aligned two column grid"
      >
        <div
          className="column"
        >
          <div
            className="ui items homeItem"
          >
            <div
              className="item"
            >
              <div
                className="image"
              >
                <i
                  aria-hidden="true"
                  className="purple wrench huge bordered inverted icon"
                  onClick={[Function]}
                />
              </div>
              <div
                className="content"
              >
                <h1
                  className="header"
                >
                  <a
                    href="/designs/shop-shelves"
                    onClick={[Function]}
                  >
                    Shop Shelves
                  </a>
                </h1>
                <div
                  className="description"
                >
                  A simple shelf design
                </div>
                <div
                  className="extra"
                >
                  <div
                    className="ui purple label"
                    onClick={[Function]}
                  >
                    Designs
                  </div>
                </div>
              </div>
            </div>
            <div
              className="item"
            >
              <div
                className="image"
              >
                <i
                  aria-hidden="true"
                  className="purple code huge bordered inverted icon"
                  onClick={[Function]}
                />
              </div>
              <div
                className="content"
              >
                <h1
                  className="header"
                >
                  <a
                    href="/editors/markdown"
                    onClick={[Function]}
                  >
                    Markdown
                  </a>
                </h1>
                <div
                  className="description"
                >
                  A split view Markdown editor which will render content as you type
                </div>
                <div
                  className="extra"
                >
                  <div
                    className="ui purple label"
                    onClick={[Function]}
                  >
                    Editors
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="column"
        >
          <div
            className="ui items homeItem"
          >
            <div
              className="item"
            >
              <div
                className="image"
              >
                <i
                  aria-hidden="true"
                  className="purple code huge bordered inverted icon"
                  onClick={[Function]}
                />
              </div>
              <div
                className="content"
              >
                <h1
                  className="header"
                >
                  <a
                    href="/editors/json"
                    onClick={[Function]}
                  >
                    JSON
                  </a>
                </h1>
                <div
                  className="description"
                >
                  A simple JSON editor with minification and pretty options
                </div>
                <div
                  className="extra"
                >
                  <div
                    className="ui purple label"
                    onClick={[Function]}
                  >
                    Editors
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
